File: examples/typescript/pagination/snapshot.ts
import { Northrays } from '@northrays/sdk'

async function main() {
  const northrays = new Northrays()

  const result = await northrays.snapshot.list(2, 10)
  console.log(`Found ${result.total} snapshots`)
  result.items.forEach((snapshot) => console.log(`${snapshot.name} (${snapshot.imageName})`))
}

main().catch(console.error)

File: examples/python/pagination/_async/snapshot.py
import asyncio

from northrays import AsyncNorthrays


async def main():
    async with AsyncNorthrays() as northrays:
        result = await northrays.snapshot.list(page=2, limit=10)
        for snapshot in result.items:
            print(f"{snapshot.name} ({snapshot.image_name})")


if __name__ == "__main__":
    asyncio.run(main())
